File: 이상욱/BOJ/m2w3/BOJ9095.java
package m2w8;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class BOJ9095 {

	static int[] ans = new int[11];

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int testNum = Integer.parseInt(br.readLine());

		ans[0] = 0;
		ans[1] = 1;
		ans[2] = 2;
		ans[3] = 4;

		for (int i = 4; i < ans.length; i++) {
			ans[i] = ans[i - 3] + ans[i - 2] + ans[i - 1];
		}

		for (int i = 0; i < testNum; i++) {

			int num = Integer.parseInt(br.readLine());

			System.out.println(ans[num]);

		}
	}
}

File: 이상욱/BOJ/m2w8/BOJ9095.kt
package m2w8

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {

    val br = BufferedReader(InputStreamReader(System.`in`))

    val n = Integer.parseInt(br.readLine())
    val arr = IntArray(11)

    arr[1] = 1
    arr[2] = 2
    arr[3]=  4

    for (i in 4 until arr.size){
        arr[i] = arr[i-1] + arr[i-2] + arr[i-3]
    }


    for (i in 0 until n){
        val num = Integer.parseInt(br.readLine())

        println(arr[num])

    }

}